src/store/item.js

```diff
@@ -31,52 +35,124 @@ const newItem = () => ({
+export const changeItemField = (field, value) => {
   item.update(itm => ({ ...itm, [field]: value }));
-}
+};
 
-export function changeVarietyField(id, field, value) {
+export const changeVarietyField = (id, field, value) => {
   item.update(itm => {
     let varieties = itm.varieties;
     varieties[id][field] = value;
     return {...itm, varieties};
   });
-}
-
-export function changeVarietyQuantity(id, value) {
-  item.update(itm => {
-    let varieties = itm.varieties;
-    varieties[id].quantity = value;
-    return {...itm, varieties};
-  });
-}
+};
 
-export function changeVarietySize(id, size, value) {
+export const changeVarietySize = (id, size, value) => {
   item.update(itm => {
     let varieties = itm.varieties;
     varieties[id].sizes[size] = value;
     return {...itm, varieties};
   });
-}
+};
 
-export function toggleSizes() {
+export const toggleSizes = () => {
   item.update(itm => ({ ...itm, inSizes: !itm.inSizes }));
-}
+  saveDraft();
+};
 
-export function addVariety() {
+export const addVariety = () => {
   item.update(itm => ({ ...itm, varieties: [...itm.varieties, newVariety()] }));
-}
+  saveDraft();
+};
 
-export function removeVariety(index) {
+export const removeVariety = (index) => {
   item.update(itm => ({ ...itm, varieties: itm.varieties.filter((x, i) => i !== index) }));
-}
+  saveDraft();
+};
+
+export const uploadFiles = async files => {
+  let images = [];
+  for (let file of files) {
+    const formData = new FormData();
+    formData.append('file', file);
+    images.push(await request(fetch, '/file/product', 'POST', formData, true));
+  }
+  return images;
+};
 
-export function clearAll() {
+export const clearAll = () => {
   item.update(() => newItem());
-}
+  saveDraft();
+};
+
+export const getDraft = () => {
+  let product = JSON.parse(localStorage.getItem('product-draft'));
+  if (product) item.update(() => product);
+  validate();
+};
+
+export const saveDraft = () => {
+  localStorage.setItem('product-draft', JSON.stringify(get(item)));
+  validate();
+};
+
+const validate = () => {
+  let product = get(item);
+
+  let error = true;
+  if (product.name && product.price && product.varieties[0].images.length) error = false;
+
+  changeItemField('error', error);
+  return error;
+};
+
+export const createProduct = async () => {
+  if (validate()) return;
+
+  const product = get(item);
+  let varieties = [];
+  product.varieties.forEach(variety => {
+    let color = variety.color.replace(/[^0-9a-f]/gi, '');
+    let imgs = variety.images.map(img => img.url);
+
+    if (product.inSizes) {
+      for (let size in variety.sizes) {
+        if (variety.sizes[size] > 0) {
+          varieties.push({
+            color: color,
+            images: imgs,
+            amount: variety.sizes[size],
+            size,
+          });
+        }
+      }
+    } else {
+      varieties.push({
+        color: variety.color.replace(/[^0-9a-f]/gi, ''),
+        images: variety.images.map(img => img.url),
```

Comment:
```suggestion
        images: imgs,
```

Reply:
Nope, backend needs only uri strings, not objects